src/components/Badge/Badge.tsx

```diff
@@ -0,0 +1,65 @@
+/* eslint-disable complexity */
+import React from 'react';
+import { withTheme } from 'styled-components/native';
+
+import {
+  Theme,
+} from '../../common/themeSelectors';
+
+import { BadgeContainer, BadgeContent } from './Badge.style';
+
+export type BadgeTypes = 'dot' | 'standard';
+export type BadgeColors = 'alert' | 'primary' | 'secondary' | 'success';
```

Comment:
Aqui é necessário importar as cores do `@natds-themes`. Você pode fazer um `Pick` pra aceitar essas 4 cores:

```suggestion
export type BadgeColors = Pick<Color, 'alert' | 'primary' | 'secondary' | 'success'>;
```